packages/notices/src/store/actions.js

```diff
@@ -47,7 +48,11 @@ export function* createNotice( status = DEFAULT_STATUS, content, options = {} )
 	content = String( content );
 
 	if ( speak ) {
-		yield { type: 'SPEAK', message: content };
+		yield {
+			type: 'SPEAK',
+			message: content,
+			ariaLive: type === 'snackbar' ? 'polite' : 'assertive',
+		};
```

Comment:
While I kept the a11y message tied to the notices package like today, I feel like a possibly better solution is to make these message as "effects" in the Notice/Snackbar components, that way you don't have to rely on the notices package for these to be announced but it would also work for the components when used directly.

I'm leaving this for a potential separate PR. cc @aduth

Reply:
> While I kept the a11y message tied to the notices package like today, I feel like a possibly better solution is to make these message as "effects" in the Notice/Snackbar components, that way you don't have to rely on the notices package for these to be announced but it would also work for the components when used directly.

That seems reasonable to me, though I'll open an issue to consider it for discussion and tracking: #15708

It would make sense if the spoken message would be considered as part of the rendered interface of the application, and not as much a side-effect of the data flow, which in retrospect seems the more sensible approach.

Reply:
> It would make sense if the spoken message would be considered as part of the rendered interface of the application, and not as much a side-effect of the data flow, which in retrospect seems the more sensible approach.

It definitely should be something you should be able to opt in or out from. I can imagine that someone could use [ARIA live region](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions) instead to announce those messages.